File: bench/prongB/binary_trees.go
// binary-trees, Go — pointer structs, garbage collected.
package main
import "fmt"
type Tree struct{ l, r *Tree }
func make_(d int) *Tree {
	t := &Tree{}
	if d > 0 {
		t.l = make_(d - 1)
		t.r = make_(d - 1)
	}
	return t
}
func check(t *Tree) int {
	if t.l == nil {
		return 1
	}
	return 1 + check(t.l) + check(t.r)
}
func main() {
	mind, maxd := 4, 18
	stretch := maxd + 1
	fmt.Printf("stretch tree of depth %d check: %d\n", stretch, check(make_(stretch)))
	longlived := make_(maxd)
	for d := mind; d <= maxd; d += 2 {
		iters := 1
		for k := 0; k < maxd-d+mind; k++ {
			iters *= 2
		}
		sum := 0
		for i := 0; i < iters; i++ {
			sum += check(make_(d))
		}
		fmt.Printf("%d trees of depth %d check: %d\n", iters, d, sum)
	}
	fmt.Printf("long-lived tree of depth %d check: %d\n", maxd, check(longlived))
}

File: bench/prongB/binary_trees.rs
// binary-trees, Rust — Box-owned recursive enum, RAII drop per tree.
enum Tree { Leaf, Node(Box<Tree>, Box<Tree>) }
fn make(d: i64) -> Tree {
    if d == 0 { Tree::Leaf } else { Tree::Node(Box::new(make(d - 1)), Box::new(make(d - 1))) }
}
fn check(t: &Tree) -> i64 {
    match t { Tree::Leaf => 1, Tree::Node(l, r) => 1 + check(l) + check(r) }
}
fn main() {
    let (mind, maxd) = (4i64, 18i64);
    let stretch = maxd + 1;
    println!("stretch tree of depth {} check: {}", stretch, check(&make(stretch)));
    let longlived = make(maxd);
    let mut d = mind;
    while d <= maxd {
        let mut iters = 1i64; for _ in 0..(maxd - d + mind) { iters *= 2; }
        let mut sum = 0i64;
        for _ in 0..iters { sum += check(&make(d)); }
        println!("{} trees of depth {} check: {}", iters, d, sum);
        d += 2;
    }
    println!("long-lived tree of depth {} check: {}", maxd, check(&longlived));
}
